README.md

```diff
@@ -31,6 +31,17 @@ The move to more consistent field names in some cases will have resulted in the
 
 The change in type for [basic types](https://tour.golang.org/basics/11) that are optional on input structs related to write/update operations is designed to avoid unexpected behaviours when dealing with their zero value (see [this reference](https://willnorris.com/2014/05/go-rest-apis-and-pointers/) for more details). As part of this change we now provide [helper functions](./fastly/basictypes_helper.go) to assist with generating the new pointer types required.
 
+> Note: some read/list operations require fields to be provided but if omitted a zero value will be used when marshaling the data structure into JSON. This too can cause confusion, which is why some input structs define their mandatory fields as pointers (to ensure that the backend can distinguish between a zero value and an omitted field).
+
+## Migrating from v2 to v3
```

Comment:
@phamann I added this to the README because I realized that these changes were potentially a breaking change for consumers of Go-Fastly so it should probably be a v3.0.0 release 😬 

That said, I don't think we should necessarily be afraid of major version releases. I know it would be nice to have a major release contain something 'juicy' but the fact is if something should be fixed for sustainability purposes then we shouldn't feel like we have to wait to bump a major version.

As a side note if we go to v3 then we could look to incorporate this PR https://github.com/fastly/go-fastly/pull/151 which I believe would be a breaking change as well.

Reply:
Makes sense 👍🏻 
Also +1 to incorporating any other breaking change PRs if we are to bump the version.